usaspending_api/search/v2/views/spending_by_award.py

```diff
@@ -0,0 +1,259 @@
+                '"award_type_groups": {{'
+                '"procurement": {},'
+                '"assistance": {}}}}}'
+            ).format(json.dumps(procurement_type_mapping), json.dumps(assistance_type_mapping))
+            raise UnprocessableEntityException(json.loads(error_msg))
+
+        subawards_values = list(contract_subaward_mapping.keys()) + list(grant_subaward_mapping.keys())
+        if self.sort not in subawards_values:
+            raise InvalidParameterException(
+                "Sort value '{}' not found in Sub-Award mappings: {}".format(self.sort, subawards_values)
+            )
+
+        queryset = subaward_filter(filters)
+        values = {"subaward_number", "piid", "fain", "award_type"}
+
+        for field in self.fields:
+            if contract_subaward_mapping.get(field):
+                values.add(contract_subaward_mapping.get(field))
+            if grant_subaward_mapping.get(field):
+                values.add(grant_subaward_mapping.get(field))
+
+        if set(filters["award_type_codes"]) <= set(procurement_type_mapping):
+            sort_filters = [contract_subaward_mapping[self.sort]]
+        elif set(filters["award_type_codes"]) <= set(assistance_type_mapping):
+            sort_filters = [grant_subaward_mapping[self.sort]]
+        else:
+            msg = "Sort key '{}' doesn't exist for the selected Award group".format(self.sort)
+            raise InvalidParameterException(msg)
+
+        if self.sort == "Award ID":
+            if self.order == "desc":
+                queryset = queryset.order_by(
+                    F("award__piid").desc(nulls_last=True), F("award__fain").desc(nulls_last=True)
+                ).values(*list(values))
+            else:
+                queryset = queryset.order_by(
+                    F("award__piid").asc(nulls_last=True), F("award__fain").asc(nulls_last=True)
+                ).values(*list(values))
+        elif self.order == "desc":
+            queryset = queryset.order_by(F(sort_filters[0]).desc(nulls_last=True)).values(*list(values))
+        else:
+            queryset = queryset.order_by(F(sort_filters[0]).asc(nulls_last=True)).values(*list(values))
+
+        limited_queryset = queryset[self.lower_bound : self.upper_bound]
+        has_next = len(limited_queryset) > self.limit
+
+        results = []
+        for award in limited_queryset[: self.limit]:
+            row = {"internal_id": award["subaward_number"]}
+
+            if award["award_type"] == "procurement":
+                for field in self.fields:
+                    row[field] = award.get(contract_subaward_mapping[field])
+            elif award["award_type"] == "grant":
+                for field in self.fields:
+                    row[field] = award.get(grant_subaward_mapping[field])
+            results.append(row)
+
+        return {"limit": self.limit, "results": results, "page_metadata": {"page": self.page, "hasNext": has_next}}
```

Comment:
In response to the comment thread on line 73, here's another example with the same format. I wouldn't suggest trying to standardize the responses across every view, just handle complete duplicates. These views all used to be the same, so I can't imagine the JSON response has diverged that much at least between the new views. Another factor to consider is that the data contract assumes that each of these views will return responses of the same format since this seperation is abstracted.

Reply:
I'll get to work making this PR even larger

Reply:
I'd be fine with seeing this in a separate PR, and I'd be happy to do the work myself, but I'm concerned that concerns of risk will come up, and nobody is going to want to retest this area. Would you be alright if I just made the changes on your branch, which would still up PR size but at least not burden you any more?

Reply:
I made changes. Might not perfectly follow the original design principles but should be a good start.